skore/src/skore/sklearn/train_test_split/train_test_split.py

```diff
@@ -114,10 +127,26 @@ class labels.
     import sklearn.model_selection
 
     new_arrays = list(arrays)
+    keys = []
     if X is not None:
         new_arrays.append(X)
+        keys += ["X"]
     if y is not None:
         new_arrays.append(y)
+        keys += ["y"]
+
+    if as_dict and arrays:
+        raise ValueError(
+            "When return_dict=True, arrays must be passed as keyword arguments.\n"
+            "Example: train_test_split(X=X, y=y, sw=sample_weight, return_dict=True)"
+        )
+    elif keyword_arrays:
```

Comment:
Since we exit the function in the above `if`
```suggestion
    
    if keyword_arrays:
```

Reply:
How did I miss that? 😞
 
Added in https://github.com/probabl-ai/skore/pull/1460/commits/21fc68c8bf9a8f6122711e9c44abd0e0e5100363